src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -19,7 +19,11 @@ BOOL OOPStackUnwinderX86::Unwind(T_CONTEXT* pContextRecord, T_KNONVOLATILE_CONTE
 
     if (pContextPointers)
     {
-        rd.pCurrentContextPointers = pContextPointers;
+        for (int i = 0; i < 7; i++)
```

Comment:
> I still wonder if fixing the places where we use the GetEbpLocation() would not be better. But since that code is shared with Windows x86 that uses a different format of REGDISPLAY that contains only the context pointers, the proper fix would likely be adding GetEbp() method to the REGDISPLAY, use that instead of *pContext->GetEbpLocation() at all places and implement it as returning pContext->pCurrentContext->Ebp for Linux and *GetEbpLocation() on Windows. On Windows, we don't have the problem we are hitting on Linux.
> However, if you don't want to spend more time on this, I'd be fine merging in your change as-is now and possibly clean it up later.

Except `GetEbpLocation` we also have calls of `SetEbpLocation` and if we change `*pContext->GetEbpLocation()` to `pContext->pCurrentContext->Ebp` we need to update ebp in 2 places: `pContext->pCurrentContext` and `pContext->pCurrentContextPointers`. Is it reasonable?

Reply:
Yes, I think that we will need to modify the following (obviously the case for the FEATURE_EH_FUNCLETS only):
```c++
 inline void   Set##reg##Location(PDWORD p##reg) { pCurrentContextPointers->reg = p##reg; }
```
to
```c++
inline void   Set##reg##Location(PDWORD p##reg) 
{
    pCurrentContextPointers->reg = p##reg;
    pCurrentContext->reg = *p##reg;
}
```

Reply:
Thank you! I've updated patch. I will test it more carefully.